packages/sdk/src/contracts/prebuilt-implementations/marketplace.ts

```diff
@@ -296,6 +296,39 @@ export class Marketplace implements UpdateableNetwork {
     return this.contractWrapper.readContract.timeBuffer();
   }
 
+  /**
+   * Get all the offers
+   *
+   * @remarks Fetch all the offers for a specified direct listing.
+   * @example
+   * ```javascript
+   * const offers = await marketplaceContract.getOffers(listingId);
+   * const firstOffer = offers[0];
+   * ```
+   *
+   * @param listingId - the id of the listing to fetch offers for 
+   */
+   public async getOffers(
+    listingId: BigNumberish,
+  ): Promise<(Offer)[]> {
+    // get all new offer events from this contract
+    const events = await this.events.getEvents("NewOffer");
```

Comment:
you can get typed event objects back here, which will help catch any type conversion issues:

`const events = await this.events.getEvents<NewOfferEventObject>("NewOffer");`